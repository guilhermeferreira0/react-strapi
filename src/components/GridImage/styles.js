import styled, { css } from 'styled-components';
import { Container as TextComponent } from '../TextComponent/styles';

export const Container = styled.div`
  ${({ theme }) => css`
    & > ${TextComponent} {
      margin-bottom: ${theme.spacings.xhuge};
    }
  `}`;

export const Grid = styled.div`
  ${({ theme }) => css`
    display: grid;
    grid-template-columns: repeat(auto-fill, minmax(280px, 1fr));
    gap: ${theme.spacings.large};
  `}`;

export const GridElement = styled.div`
  ${({ theme }) => css`
    overflow: hidden;
    border-radius: 8px;
  `}`;

export const Image = styled.img`
  ${({ theme }) => css`
    width: 100%;
    transition: all .5s ease-in-out;
    cursor: pointer;
    border-radius: 8px;

    &:hover {
      transform: scale(1.4) rotate(10deg);
    }
  `}`;
