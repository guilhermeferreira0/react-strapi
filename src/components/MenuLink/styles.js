import styled, { css } from 'styled-components';

export const Container = styled.a`
  ${({ theme }) => css`
    display: block;
    text-decoration: none;
    font-size: ${theme.fonts.sizes.small};
    padding: ${theme.spacings.small};
    color: ${theme.colors.primaryColor};
    position: relative;
    scale: 1;
    transition: scale 0.5s ease-in-out;


    &::after {
      content: '';
      position: absolute;
      bottom: 0.75rem;
      left: 50%;
      width: 0%;
      height: 0.2rem;
      background-color: ${theme.colors.secondaryColor};
      transition: all 0.5s ease-in-out;
    }

    &:hover::after {
      left: 25%;
      width: 50%;
    }

    &:hover {
      scale: 1.1;
    }

    @media ${theme.media.ltemedium} {
      &:hover::after {
      left: 20.5%;
      width: 50%;
    }
    }
  `}`;
