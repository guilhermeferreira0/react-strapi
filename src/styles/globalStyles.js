import { createGlobalStyle } from 'styled-components';

export const GlobalStyles = createGlobalStyle`
  @import url();

  * {
    margin: 0;
    padding: 0;
    box-sizing: border-box;
    outline: none;
  }

  html {
    font-size: 62.5%;
  }

  body {
    font-size: 2.4rem; // 1
    font-family: ${({ theme }) => theme.fonts.family.default};
  }

  h1, h2, h3, h4, h5, h6 {
    font-family: ${({ theme }) => theme.fonts.family.secondary};
    font-weight: 800;
  }
`;
