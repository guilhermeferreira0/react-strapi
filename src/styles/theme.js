export const theme = {
  colors: {
    primaryColor: 'lightgray',
    secondaryColor: '#dc143c',
    white: '#FFFFFF',
    lightBg: '#FFFFFF',
    darkBg: '#0A1128',
  },
  fonts: {
    family: {
      default: "'Open Sans', sans-serif",
      secondary: "'Montserrat', sans-serif",
    },
    sizes: {
      xsmall: '0.8rem',
      small: '1.6rem',
      medium: '2.4rem',
      large: '3.2rem',
      xlarge: '4rem',
      xxlarge: '4.8rem',
      huge: '5.6rem',
      xhuge: '6.2rem',
    },
  },
  spacings: {},
};
