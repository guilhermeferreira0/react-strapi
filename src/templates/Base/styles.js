import styled, { css } from 'styled-components';

export const Container = styled.div`
  ${({ theme }) => css`
    margin-top: 8%;
    width: 100%;
    height: 100%;
  `}`;
