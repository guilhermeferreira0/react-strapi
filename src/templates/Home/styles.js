import styled, { css } from 'styled-components';

export const Wrapper = styled.div`
  ${(props) => css`
    background-color: ${props.background};
  `}
`;
