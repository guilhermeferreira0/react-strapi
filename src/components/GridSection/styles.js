import styled, { css } from 'styled-components';
import { Title as HeadingContainer } from '../Heading/styles';
import { Container as TextComponent } from '../TextComponent/styles';

export const Container = styled.div`
  ${({ theme }) => css`
    ${TextComponent} {
      margin-bottom: ${theme.spacings.xhuge};
    }
  `}`;

export const Grid = styled.div`
  ${({ theme }) => css`
    counter-reset: grid-counter;
    display: grid;
    grid-template-columns: repeat(auto-fill, minmax(280px, 1fr));
    gap: ${theme.spacings.large};

    @media ${theme.media.ltemedium} {
      grid-template-columns: 1fr;
    }
  `}`;

export const GridElement = styled.div`
  ${({ theme }) => css`
    ${HeadingContainer} {
      position: relative;
      left: 7rem;
    }

    ${HeadingContainer}::before {
      content: counter(grid-counter);
      counter-increment: grid-counter;
      position: absolute;
      font-size: ${theme.fonts.sizes.huge};
      top: -2rem;
      left: -7rem;
      transform: rotate(10deg);
    }
  `}`;
