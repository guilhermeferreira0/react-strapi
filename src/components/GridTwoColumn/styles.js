import styled, { css } from 'styled-components';
import { Title } from '../Heading/styles';

export const Container = styled.div`
  ${({ theme }) => css`
    display: grid;
    grid-template-columns: 1fr 2fr;
    align-items: center;
    gap: ${theme.spacings.large};

    @media ${theme.media.ltemedium} {
      grid-template-columns: 1fr;
      text-align: center;
    }

    ${Title} {
      margin-bottom: ${theme.spacings.xlarge};
    }
  `}`;

export const TextContainer = styled.div`
  ${({ theme }) => css`
    @media ${theme.media.ltemedium} {
      margin-bottom: ${theme.spacings.large};
    }
  `}`;

export const ImageContainer = styled.div`
  ${({ theme }) => css``}`;

export const Image = styled.div`
  ${({ theme }) => css``}`;
