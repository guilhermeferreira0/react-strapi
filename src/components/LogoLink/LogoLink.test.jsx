import { screen } from '@testing-library/react';
import { LogoLink } from '.';
import { renderTheme } from '../../styles/renderTheme';

describe('<LogoLink />', () => {
  it('Should render text logo', () => {
    renderTheme(<LogoLink link="#target" text="olá mundo" />);
    const heading = screen.getByRole('heading', { name: 'olá mundo' });
    expect(heading.firstChild).toHaveAttribute('href', '#target');
  });

  // it('Should render image logo', () => {
  //   renderTheme(
  //     <LogoLink link="#target" text="olá mundo" srcImage="image.jpg" />,
  //   );
  //   const heading = screen.getByRole('heading', { name: 'olá mundo' });
  //   expect(heading.firstChild.firstChild).toHaveAttribute('src', 'image.jpg');
  // });

  it('Should render image logo', () => {
    renderTheme(
      <LogoLink link="#target" text="olá mundo" srcImage="image.jpg" />,
    );
    expect(screen.getByRole('img', { name: 'olá mundo' })).toHaveAttribute(
      'src',
      'image.jpg',
    );
  });

  it('Should match snapshot', () => {
    const { container } = renderTheme(
      <LogoLink link="#target" text="olá mundo" srcImage="image.jpg" />,
    );

    expect(container.firstChild).toMatchSnapshot();
  });
});
