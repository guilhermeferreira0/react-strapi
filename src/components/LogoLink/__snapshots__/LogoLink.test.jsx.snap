// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LogoLink /> Should match snapshot 1`] = `
.c1 {
  display: flex;
  align-items: center;
  text-decoration: none;
  color: inherit;
}

.c1>img {
  max-height: 3rem;
}

.c0 {
  color: black;
  font-size: 2.4rem;
  text-transform: uppercase;
}

<h1
  class="c0"
>
  <a
    class="c1"
    href="#target"
  >
    <img
      alt="olÃ¡ mundo"
      src="image.jpg"
    />
  </a>
</h1>
`;
